src/index.js

```diff
@@ -85,28 +84,377 @@ const File = require('./file.js');
  *
  * @param {ClientConfig} [options] Configuration options.
  */
-function Storage(options) {
-  if (!(this instanceof Storage)) {
-    return new Storage(options);
+class Storage extends common.Service {
+  constructor(options) {
+    const config = {
+      baseUrl: 'https://www.googleapis.com/storage/v1',
+      projectIdRequired: false,
+      scopes: [
+        'https://www.googleapis.com/auth/iam',
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/devstorage.full_control',
+      ],
+      packageJson: require('../package.json'),
+    };
+
+    super(config, options);
+
+    /**
+     * Reference to {@link Storage.acl}.
+     *
+     * @name Storage#acl
+     * @see Storage.acl
+     */
+    this.acl = Storage.acl;
```

Comment:
Object properties will be defined in the class directly (not from constructor) when we convert to Typescript.